test/helpers.py

```diff
@@ -281,26 +296,40 @@ def sde_solver_strong_order(
     return steps_arr, errs_arr, order
 
 
-@dataclasses.dataclass(frozen=True)
+@dataclasses.dataclass(unsafe_hash=True)
 class SDE:
     get_terms: Callable[[AbstractBrownianPath], AbstractTerm]
     args: PyTree
     y0: PyTree[Array]
     t0: float
     t1: float
-    w_shape: tuple[int, ...]
+    _w_shape: Union[tuple[int, ...], PyTree[jax.ShapeDtypeStruct]]
+
+    @property
+    def w_shape(self):
+        if is_tuple_of_ints(self._w_shape):
+            return self._w_shape
+        else:
+            return self._w_shape
```

Comment:
?

Reply:
A complete mystery to me as well. I did get a good laugh out of it though. And why I needed it to be a property is also beyond me.